archinstall/lib/disk/partition.py

```diff
@@ -171,26 +171,32 @@ def __dump__(self) -> Dict[str, Any]:
 
 	def _call_lsblk(self) -> Dict[str, Any]:
 		self.partprobe()
-		# This sleep might be overkill, but lsblk is known to
-		# work against a chaotic cache that can change during call
-		# causing no information to be returned (blkid is better)
-		# time.sleep(1)
-
-		# TODO: Maybe incorporate a re-try system here based on time.sleep(max(0.1, storage.get('DISK_TIMEOUTS', 1)))
-
-		try:
-			output = SysCommand(f"lsblk --json -b -o+LOG-SEC,SIZE,PTTYPE,PARTUUID,UUID,FSTYPE {self.device_path}").decode('UTF-8')
-		except SysCallError as error:
-			# It appears as if lsblk can return exit codes like 8192 to indicate something.
-			# But it does return output so we'll try to catch it.
-			output = error.worker.decode('UTF-8')
-
-		if output:
+		for _ in range(storage.get('DISK_RETRY_ATTEMPTS', 3)):
+			try:
+				output = SysCommand(f"lsblk --json -b -o+LOG-SEC,SIZE,PTTYPE,PARTUUID,UUID,FSTYPE {self.device_path}").decode('UTF-8')
+			except SysCallError as error:
+				# It appears as if lsblk can return exit codes like 8192 to indicate something.
+				# But it does return output so we'll try to catch it.
+				log(f"Error running lsblk: {error.worker.decode('UTF-8')}", level=logging.DEBUG)
+				# Catch the error but try again anyway
+				delay = max(0.1, storage.get('DISK_TIMEOUTS', 1))
+				log(f"Waiting {delay}s to poll disk again", level=logging.DEBUG)
+				time.sleep(delay)
+				continue
+			
 			try:
 				lsblk_info = json.loads(output)
-				return lsblk_info
 			except json.decoder.JSONDecodeError:
 				log(f"Could not decode JSON: {output}", fg="red", level=logging.ERROR)
+				# This is unexpected, better to raise an error
+				break
+			if not lsblk_info['blockdevices'][0]['partuuid']:
```

Comment:
Could we use `lsblk_info.get('blockdevices', [{'partuuid' None}])[0].get('partuuid')` here? To avoid `IndexError` and `KeyError` when the path doesn't exist?

Reply:
Of course